### ISSUE TITLE
Faster sbt new

### ISSUE THREAD

**Issue by laughedelic** [sbt/sbt#3391]
The `sbt new` command is great because it makes it easy to start new projects by bringing giter8 to a wider audience. But it's far away from being an alias for giter8. 

* it takes it ~15s to load before it starts asking template-related questions (comparing to ~3s of the pure giter8)
* it creates `target/` directory where it's not needed (I know it's mentioned already in #2835, so I'm not going to repeat anything about it here)

I don't know much about sbt internals, so probably there's no way to change this behavior, but let me ask some questions:

* I think that the most common use case for `sbt new` is creating a new project from outside of any existing sbt projects (because it creates a new directory). Is there any use case in which loading current dir as a project is justified? Is it possible to load sbt without any project?
* I wanted to ask a similar question about global plugins loading, but I suppose that loading global plugins allows users to add different template resolvers in addition to giter8. So I guess, it's unavoidable, right?
* Would it be possible to make it apply the template in the _current_ directory? Kind of `sbt init` (similar to `git init` and some other tools).

In summary I just want to know how possible it is to make `sbt new` fast and lightweight, bypassing project loading and whatever else is happening under the hood which takes it so long.

**Comment by eed3si9n** [comment]
`new` is implemented as a command that is configurable via the `templateResolverInfos` setting key, so we need to load the current project.

> Is it possible to load sbt without any project?

When we know that we don't have any `build.sbt` or `project` directory, maybe there's some shortcut we can try to make to cut down the loading time. 

> I wanted to ask a similar question about global plugins loading, but I suppose that loading global plugins allows users to add different template resolvers in addition to giter8. So I guess, it's unavoidable, right?

I guess so.

> Would it be possible to make it apply the template in the current directory? Kind of sbt init (similar to git init and some other tools).

sbt new just delegates to Giter8 here, so it's more of a question about Giter8 supporting this feature. I just created https://github.com/foundweekends/giter8/issues/330.

**Comment by laughedelic** [comment]
> When we know that we don't have any build.sbt or project directory, maybe there's some shortcut we can try to make to cut down the loading time.

This would be wonderful.

> sbt new just delegates to Giter8 here, so it's more of a question about Giter8 supporting this feature. I just created foundweekends/giter8#330.

OK, subscribed there 👍
